keps/sig-auth/3766-referential-authorization/README.md

```diff
@@ -413,80 +429,121 @@ rejected".
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // From refers to the group, resource, and pthat that these references
+    // originate from.
+    From    GroupVersionResourcePath `json:"from"`
+
+    // To refers to the group and resource that these references target.
+    To      GroupResource `json:"to"`
+
+    // For refers to the purpose of this reference. Subjects of
+    // ClusterReferenceConsumers will be authorized to follow references matching
+    // the From, To, and For of this resource.
+    //
+    // This value must be a valid DNS label as defined per RFC-1035.
+    For     string        `json:"for"`
 }
 
-// +kubebuilder:object:root=true
-// ReferenceGrantList contains a list of ReferenceGrant.
-type ReferenceGrantList struct {
-	metav1.TypeMeta `json:",inline"`
-	metav1.ListMeta `json:"metadata,omitempty"`
-	Items           []ReferenceGrant `json:"items"`
+type GroupVersionResourcePath struct {
+    Group    string `json:"group"`
+    Resource string `json:"resource"`
+    Version  string `json:"version"`
+    // JSONPath is the JSONPath which this reference may come from.
+    JSONPath string `json:"jsonPath"`
 }
+```
 
-// ReferenceGrantSpec identifies a cross namespace relationship that is trusted
-// for Gateway API.
-type ReferenceGrantSpec struct {
-	// From describes the trusted namespaces and kinds that can reference the
-	// resources described in "To". Each entry in this list MUST be considered
-	// to be an additional place that references can be valid from, or to put
-	// this another way, entries MUST be combined using OR.
-	//
-	// +kubebuilder:validation:MinItems=1
-	// +kubebuilder:validation:MaxItems=16
-	From []ReferenceGrantFrom `json:"from"`
-
-	// To describes the resources in this namespace that may be referenced by
-	// the resources described in "From". Each entry in this list MUST be
-	// considered to be an additional set of objects that references can be
-	// valid to, or to put this another way, entries MUST be combined using OR.
-	//
-	// +kubebuilder:validation:MinItems=1
-	// +kubebuilder:validation:MaxItems=16
-	To []ReferenceGrantTo `json:"to"`
+#### ReferenceGrant
+
+```golang
+// ReferenceGrant identifies namespaces of resources that are trusted to
+// reference the specified names of resources in the same namespace as the
+// grant.
+type ReferenceGrant struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // From refers to the group and resource that these references originate from.
+    From    GroupResource `json:"from"`
+
+    // To refers to the group, resource, and names that these references can
+    // target.
+    To      ReferenceGrantTo `json:"to"`
+
+    // For refers to the purpose of this reference. Subjects of
+    // ClusterReferenceConsumers will be authorized to follow references
+    // matching the From, To, and For of this resource.
+    //
+    // This value must be a valid DNS label as defined per RFC-1035.
```

Comment:
```suggestion
    // This value must be a valid DNS label as defined per RFC-1035, and is matched case-sensitively.
```

Reply:
Can we have this as a case-insenstive match?